classes/autoupdate.php

```diff
@@ -85,6 +85,7 @@ public static function update_from_delete_event(\core\event\base $event): void {
                         foreach ($placestore->places as $p) {
                             if ($p->linkedActivity == $data['objectid']) {
                                 $p->linkedActivity = null;
+                                cachemanager::remove_cmid($data['objectid']);
```

Comment:
why do you need this when you are rebuilding the whole course cache anyway a few lines below that?

I would suggest to do only one of them: Either remove a course module from the cache, but not rebuild the cache for the whole course, or just wipe the course cache and rebuild it.

Wiping the whole course cache however could be problematic: Imagine a batch delete or course delete (which basically results in a batch delete of the course modules). While deleting a course with 100 modules this rebuilds the course cache up to 100 times.

On the other hand, if we do not purge the course cache here, we need an additional mechanism in case the course module which is being deleted, is a learningmap instance with backlinks enabled. In this case the cache needs to be manipulated to remove the entries pointing to this learningmap instance.

Reply:
There was sth missing in autoupdate::reset_backlink_cache(): This should only be run, if the updated / deleted module was a learningmap. I added the missing lines there.